src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/client/BaseHMPPSClient.kt

```diff
@@ -86,6 +81,37 @@ abstract class BaseHMPPSClient(
     }
   }
 
+  private fun <ResponseType : Any> performRequest(typeReference: TypeReference<ResponseType>, method: HttpMethod, requestBuilder: HMPPSRequestConfiguration): Pair<HttpStatus, ResponseType> {
+    val request = webClient.method(method)
+      .uri(requestBuilder.path ?: "")
+      .headers { it.addAll(requestBuilder.headers) }
+
+    if (requestBuilder.body != null) {
+      request.bodyValue(requestBuilder.body!!)
+    }
+
+    val result = request
+      .retrieve()
+      .toEntityFlux(DataBuffer::class.java)
+      .block()!!
+
+    val pipedOutputStream = PipedOutputStream()
+    val pipedInputStream = PipedInputStream(pipedOutputStream)
+
+    val body = result
+      .body
+      ?.doOnError { pipedInputStream.close() }
+      ?.doFinally { pipedOutputStream.close() }
+
+    DataBufferUtils
```

Comment:
This should help us on the large query for offender summaries though I worry what the effect might be for all/most other requests. Is the overhead of setting this up each time for every small request worth the trade off? Are there other trade offs?

Could/should we introduce a parameter to stream only specific requests?

Reply:
I did consider doing this, but I'm inclined to think this extra complexity wouldn't be worth it from a performance perspective, as I suspect that we'd only see a very small overhead if any, and there's a good chance that constructing the `String` instance as an intermediate step and passing that to the `ObjectMapper` would negate the benefits from skipping that overhead. Also, I suspect that internally something like this happens anyway, but just for a single buffer (<250kB/750kB depending on configuration).